components/enterprise-contract/download-service.yaml

```diff
@@ -0,0 +1,234 @@
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: ec-cli-download-role
+  namespace: enterprise-contract-service
+rules:
+  - apiGroups:
+      - route.openshift.io
+    resourceNames:
+      - ec-cli-download-route
+    resources:
+      - routes
+    verbs:
+      - get
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name: ec-cli-download-role-binding
+  namespace: enterprise-contract-service
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: ec-cli-download-role
+subjects:
+  - kind: ServiceAccount
+    name: default
+    namespace: enterprise-contract-service
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: ec-cli-download-cluster-role
+  namespace: enterprise-contract-service
+rules:
+  - apiGroups:
+      - console.openshift.io
+    resources:
+      - consoleclidownloads
+    verbs:
+      - create
+      - patch
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: ec-cli-download-cluster-role-binding
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: ec-cli-download-cluster-role
+subjects:
+  - kind: ServiceAccount
+    name: default
+    namespace: enterprise-contract-service
+---
+apiVersion: apps/v1
+kind: Deployment
```

Comment:
This deployment is is adding 13 kube-linter issues. This should be detected at the PR level but we did not enable it yet, we need to fix all existing issue before we can enable it. Can you address those please:
```
kustomize build components/enterprise-contract/ | kube-linter lint -
KubeLinter v0.6.4-0-g4f0b1e01fd

<standard input>: (object: enterprise-contract-service/ec-cli-download-deployment apps/v1, Kind=Deployment) The container "download" is using an invalid container image, "image-registry.openshift-image-registry.svc:5000/openshift/tools:latest". Please use images that are not blocked by the `BlockList` criteria : [".*:(latest)$" "^[^:]*$" "(.*/[^:]+)$"] (check: latest-tag, remediation: Use a container image with a specific tag other than latest.)

<standard input>: (object: enterprise-contract-service/ec-cli-download-deployment apps/v1, Kind=Deployment) container "download" does not have a read-only root file system (check: no-read-only-root-fs, remediation: Set readOnlyRootFilesystem to true in the container securityContext.)

<standard input>: (object: enterprise-contract-service/ec-cli-download-deployment apps/v1, Kind=Deployment) container "http" does not have a read-only root file system (check: no-read-only-root-fs, remediation: Set readOnlyRootFilesystem to true in the container securityContext.)

<standard input>: (object: enterprise-contract-service/ec-cli-download-deployment apps/v1, Kind=Deployment) container "download" is not set to runAsNonRoot (check: run-as-non-root, remediation: Set runAsUser to a non-zero number and runAsNonRoot to true in your pod or container securityContext. Refer to https://kubernetes.io/docs/tasks/configure-pod-container/security-context/ for details.)

<standard input>: (object: enterprise-contract-service/ec-cli-download-deployment apps/v1, Kind=Deployment) container "http" is not set to runAsNonRoot (check: run-as-non-root, remediation: Set runAsUser to a non-zero number and runAsNonRoot to true in your pod or container securityContext. Refer to https://kubernetes.io/docs/tasks/configure-pod-container/security-context/ for details.)

<standard input>: (object: enterprise-contract-service/ec-cli-download-deployment apps/v1, Kind=Deployment) container "download" has cpu request 0 (check: unset-cpu-requirements, remediation: Set CPU requests and limits for your container based on its requirements. Refer to https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/#requests-and-limits for details.)

<standard input>: (object: enterprise-contract-service/ec-cli-download-deployment apps/v1, Kind=Deployment) container "download" has cpu limit 0 (check: unset-cpu-requirements, remediation: Set CPU requests and limits for your container based on its requirements. Refer to https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/#requests-and-limits for details.)

<standard input>: (object: enterprise-contract-service/ec-cli-download-deployment apps/v1, Kind=Deployment) container "http" has cpu request 0 (check: unset-cpu-requirements, remediation: Set CPU requests and limits for your container based on its requirements. Refer to https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/#requests-and-limits for details.)

<standard input>: (object: enterprise-contract-service/ec-cli-download-deployment apps/v1, Kind=Deployment) container "http" has cpu limit 0 (check: unset-cpu-requirements, remediation: Set CPU requests and limits for your container based on its requirements. Refer to https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/#requests-and-limits for details.)

<standard input>: (object: enterprise-contract-service/ec-cli-download-deployment apps/v1, Kind=Deployment) container "download" has memory request 0 (check: unset-memory-requirements, remediation: Set memory requests and limits for your container based on its requirements. Refer to https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/#requests-and-limits for details.)

<standard input>: (object: enterprise-contract-service/ec-cli-download-deployment apps/v1, Kind=Deployment) container "download" has memory limit 0 (check: unset-memory-requirements, remediation: Set memory requests and limits for your container based on its requirements. Refer to https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/#requests-and-limits for details.)

<standard input>: (object: enterprise-contract-service/ec-cli-download-deployment apps/v1, Kind=Deployment) container "http" has memory request 0 (check: unset-memory-requirements, remediation: Set memory requests and limits for your container based on its requirements. Refer to https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/#requests-and-limits for details.)

<standard input>: (object: enterprise-contract-service/ec-cli-download-deployment apps/v1, Kind=Deployment) container "http" has memory limit 0 (check: unset-memory-requirements, remediation: Set memory requests and limits for your container based on its requirements. Refer to https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/#requests-and-limits for details.)

Error: found 13 lint errors

```

Reply:
Thanks, I needed to add two ignores, so with that the lint issues should be resolved.

Reply:
Thanks